src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/Hashing.cs

```diff
@@ -138,14 +141,17 @@ public static unsafe int GetHashCodeOrdinalIgnoreCaseAscii(ReadOnlySpan<char> s)
 
         public static unsafe int GetHashCodeOrdinalIgnoreCase(ReadOnlySpan<char> s)
         {
+#if NET
+            return string.GetHashCode(s, StringComparison.OrdinalIgnoreCase);
+#else
             int length = s.Length;
 
             char[]? rentedArray = null;
             Span<char> scratch = length <= 256 ?
                 stackalloc char[256] :
                 (rentedArray = ArrayPool<char>.Shared.Rent(length));
 
-            length = s.ToUpperInvariant(scratch); // NOTE: this really should be the (non-existent) ToUpperOrdinal
+            length = s.ToUpperInvariant(scratch); // NOTE: this both allocates and really should be the (non-existent) ToUpperOrdinal
```

Comment:
I'm not sure how to interpret that comment. Is it suggesting that the current conversion is incorrect or that we should be adding `ToUpperOrdinal` to netfx? Should it be removed?

Reply:
I believe it's highlighting that it's trying to implement StringComparer.OrdinalIgnoreCase behavior, but it lacks the ability to do so due to not having a "upper case this text with an Ordinal mapping" and so is settling for "upper case this text with an Invariant mapping".